src/util/parented_ptr.h

```diff
@@ -104,6 +88,36 @@ inline parented_ptr<T> make_parented(Args&&... args) {
     return parented_ptr<T>(new T(std::forward<Args>(args)...));
 }
 
+template<typename T, typename U>
+inline bool operator== (const T* lhs, const parented_ptr<U>& rhs) {
+    return lhs == rhs.get();
+}
+
+template<typename T, typename U>
+inline bool operator== (const parented_ptr<T>& lhs, const T* rhs) {
+    return lhs.get() == rhs;
+}
+
+template<typename T>
+inline bool operator== (const parented_ptr<T>& lhs, const parented_ptr<T>& rhs) const {
```

Comment:
Can we get rid of the uni-type variants by providing only the dual-type template functions?